test/api/route/messages_test.dart

```diff
@@ -364,11 +371,45 @@ void main() {
       return FakeApiConnection.with_(zulipFeatureLevel: 173, (connection) async {
         await checkSendMessage(connection,
           destination: DmDestination(userIds: userIds), content: content,
+          readBySender: true,
           expectedBodyFields: {
             'type': 'private',
             'to': jsonEncode(userIds),
             'content': content,
-          });
+            'read_by_sender': 'true',
+          },
+          expectedUserAgent: 'ZulipMobile/flutter');
+      });
+    });
+
+    test('when readBySender is null, sends a User-Agent we know the server will recognize', () {
+      return FakeApiConnection.with_((connection) async {
+        await checkSendMessage(connection,
+          destination: StreamDestination(streamId, topic), content: content,
+          expectedBodyFields: {
+            'type': 'stream',
+            'to': streamId.toString(),
+            'topic': topic,
+            'content': content,
+          },
+          readBySender: null,
+          expectedUserAgent: 'ZulipMobile/flutter');
```

Comment:
nit:
```suggestion
          readBySender: null,
          expectedBodyFields: {
            'type': 'stream',
            'to': streamId.toString(),
            'topic': topic,
            'content': content,
          },
          expectedUserAgent: 'ZulipMobile/flutter');
```

because `readBySender` specifies what to do, and so comes logically before what to expect from the result